exporter/metric/prometheus/prometheus.go

```diff
@@ -0,0 +1,384 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"context"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"net/http"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"go.opentelemetry.io/otel/api/core"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+)
+
+type metricKey struct {
+	desc    *export.Descriptor
+	encoded string
+}
+
+// Exporter is an implementation of metric.Exporter that sends metrics to
+// Prometheus.
+type Exporter struct {
+	sync.RWMutex
+
+	labelEncoder export.LabelEncoder
```

Comment:
nit: somehow seems like too many newlines in this struct definition

Reply:
(I'm happy with the whitespace, fwiw.)

Reply:
I was actually not very pleased with the number of maps in a single struct, that's why I have splitted it into logical chunks. But I've refactored a bit to split into smaller structs by each type and removed these new lines.